pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -478,7 +478,7 @@ func (facts *PipelineRunFacts) GetPipelineTaskStatus() map[string]string {
 			case t.isSuccessful():
 				s = v1.TaskRunReasonSuccessful.String()
 			// execution status is Failed when a task has succeeded condition with status set to false
-			case t.isConditionStatusFalse():
+			case t.haveAnyRunsFailed():
```

Comment:
why can't this be replaced with `t.isFailure`?

Reply:
right now t.IsFailure is a stronger condition than t.haveAnyRunsFailed. IsFailure is true if any runs have failed, and all other runs are done. I suspect we could probably merge the two, and consider a pipeline task failed if any runs have failed, regardless of whether the others are done, but this seems like it would be a behavior change and this is just meant to be a refactoring. Maybe I can reopen https://github.com/tektoncd/pipeline/issues/6628 (or create a new issue) for the task of consolidating them?